junit5-engine/src/main/java/org/junit/gen5/engine/junit5/descriptor/MethodTestDescriptor.java

```diff
@@ -57,6 +54,17 @@
 	}
 
 	@Override
+	public String getUniqueId() {
+		// @formatter:off
+		return String.format("%s:%s#%s(%s)",
+				getTestEngine().getId(),
+				getTestClass().getName(),
+				getTestMethod().getName(),
+				ObjectUtils.nullSafeToString(getTestMethod().getParameterTypes()));
+		// @formatter:on
+	}
```

Comment:
Is this correct for methods in nested non-static classes?

Here and in the other `getUniqueId()` methods: Have you considered implementing this like the following (pseudo-code)?
```java
@Override
public String getUniqueId() {
	// @formatter:off
	return String.format("%s#%s(%s)",
			getParent().map(TestDescriptor::getUniqueId).orElse("")
			getTestMethod().getName(),
			ObjectUtils.nullSafeToString(getTestMethod().getParameterTypes()));
	// @formatter:on
}
```